platform/windows/os_windows.cpp

```diff
@@ -1166,8 +1166,14 @@ String OS_Windows::get_environment(const String &p_var) const {
 	return "";
 }
 
-bool OS_Windows::set_environment(const String &p_var, const String &p_value) const {
-	return (bool)SetEnvironmentVariableW((LPCWSTR)(p_var.utf16().get_data()), (LPCWSTR)(p_value.utf16().get_data()));
+void OS_Windows::set_environment(const String &p_var, const String &p_value) const {
+	ERR_FAIL_COND_MSG(p_var.is_empty() || p_var.contains("="), vformat("Invalid environment variable name '%s', cannot be empty or include '='.", p_var));
+	ERR_FAIL_COND_V_MSG(p_value.length() > 32767, vformat("Invalid value for environment variable '%s', cannot exceed 32767 characters.", p_var));
```

Comment:
Might need testing @bruvzg, I just based this off the docs.

Reply:
Probably should `p_value.utf16().length()`, it's `32,767 characters` in docs, and usually it this is the number of UTF-16 code points. Also, it's probably a combined limit for both name and value + "=" + null terminator (`Var1=Value1\0`).

Reply:
Maybe something like this to avoid double conversion and string concat.

```
	ERR_FAIL_COND_MSG(p_var.is_empty() || p_var.contains("="), vformat("Invalid environment variable name '%s', cannot be empty or include '='.", p_var));
	Char16String var = p_var.utf16();
	Char16String value = p_value.utf16();
	ERR_FAIL_COND_MSG(var.length() + value.leght() + 2 > 32767, vformat("Invalid definition for environment variable '%s', cannot exceed 32767 characters.", p_var));
	SetEnvironmentVariableW((LPCWSTR)(var.get_data()), (LPCWSTR)(value.get_data()));
```

Reply:
Thanks, used that and improved the doc as I could.

> On Windows, there is a 32767 characters limit for the combined length of [param variable], [param value], and the [code]=[/code] and null terminator characters that will be registered in the environment block.

I think we have better docs than MSDN for this now :P